### ISSUE TITLE
Kapitan error when prune: false

### ISSUE THREAD

**Issue by ademariag** [deepmind/kapitan#443]
**Describe the bug/feature**
When I go from prune: true to prune: false, compilation suddenly starts failing

**To Reproduce**
1. Remove `.kapitan` file or turn `compile: prune: false`
2. Kapitan compile
3. Observe failure

**Expected behavior**
Compilation succeedes

**Screenshots**

```
Compiled X (0.42s)
Compiled Y (0.24s)
Compiled Z (0.22s)
<some targets compile successfully>

Unknown (Non-Kapitan) Error occured
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/opt/venv/lib/python3.7/site-packages/kapitan/targets.py", line 383, in compile_target
    input_compiler.compile_obj(comp_obj, ext_vars, **kwargs)
  File "/opt/venv/lib/python3.7/site-packages/kapitan/inputs/base.py", line 47, in compile_obj
    self.compile_input_path(input_path, comp_obj, ext_vars, **kwargs)
  File "/opt/venv/lib/python3.7/site-packages/kapitan/inputs/base.py", line 67, in compile_input_path
    target_name=target_name, **kwargs)
  File "/opt/venv/lib/python3.7/site-packages/kapitan/inputs/jsonnet.py", line 72, in compile_file
    fp.write_yaml(item_value)
  File "/opt/venv/lib/python3.7/site-packages/kapitan/inputs/base.py", line 119, in write_yaml
    yaml.dump_all(obj, stream=self.fp, indent=indent, Dumper=PrettyDumper, default_flow_style=False)
  File "/opt/venv/lib/python3.7/site-packages/yaml/__init__.py", line 277, in dump_all
    for data in documents:
TypeError: 'NoneType' object is not iterable
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.7/site-packages/kapitan/targets.py", line 94, in compile_targets
    [p.get() for p in pool.imap_unordered(worker, target_objs) if p]
  File "/opt/venv/lib/python3.7/site-packages/kapitan/targets.py", line 94, in <listcomp>
    [p.get() for p in pool.imap_unordered(worker, target_objs) if p]
  File "/usr/local/lib/python3.7/multiprocessing/pool.py", line 748, in next
    raise value
TypeError: 'NoneType' object is not iterable


'NoneType' object is not iterable
```

** If it's a bug (please complete the following information):**
using docker image `eu.gcr.io/antha-images/kapitan:0.25.0-ci`

**Additional context**

**Comment by ademariag** [comment]
I have managed to identify the target which is failing. 
I do not see anything obviously wrong with it.

**Comment by uberspot** [comment]
I couldn't replicate this locally with a couple of kapitan projects.
I tried setting `compile: prune: false` in `.kapitan` and used version `0.25.4`.  
Version 0.26.0 also worked.
Can you provide more details for replication?

**Comment by ademariag** [comment]
@uberspot hitting this again. @ramaro FYI.

To replicate, just create a jsonnet file with:
```
{}
```

A more complete example is
```
local object = {};

{
  [key]: object[key]
  for key in std.objectFields(object)
}
```

The condition happens happens often if the object one is looping over is missing. i.e. 
```
local kap = import 'lib/kap.libsonnet';
local utils = kap.utils;
local p = kap.parameters;

local ingresses = utils.objectGet(p, 'ingresses', {});

{
  [ingress_name + '-ingress']:
  local ingress = ingresses[ingress_name];
  kap.K8sIngress(ingress_name)
    .WithAnnotations(utils.objectGet(ingress, 'annotations', {}))
    .WithNamespace(p.namespace)
    .WithDefaultBackend(utils.objectGet(ingress, 'backend', {}))
    .WithPaths(utils.objectGet(ingress, 'paths', {}))
  for ingress_name in std.objectFields(ingresses)
}
```
In this case, if no ingresses are defined, it will crash.

Note that adding a `+ { fixme: {} }` is currently the only way to work around the error.

The correct behaviour would be to skip the file, rather than crashing

**Comment by adrianchifor** [comment]
@ademariag This should be fixed in 0.27.2, once it's released by CI. Please re-open the issue if it's still happening after.
